src/cluster.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import * as MapboxGL from 'mapbox-gl';
+import { Props as MarkerProps } from './marker';
+
+const supercluster = require('supercluster'); //tslint:disable-line
```

Comment:
Here `require` is used to get around `noImplicitAny`(and tslint).
I think it's better to write something like this (in a global.d.ts file): 
```
declare module 'supercluster' {
    const supercluster: any;
    export default supercluster;
}
```
(Repeat this comment for each `require` like this)